src/drivers/can.rs

```diff
@@ -0,0 +1,1502 @@
+            let rx_status = self.read_register(MCP2515Register::CANINTF, 0x00).unwrap();
+            match rx {
+                RXBN::RXB0 => {
+                    if (rx_status & (1 << 0) != 0) {
+                        defmt::info!("Frame received in RXB0");
+                        let received_frame = self.read_rx_buffer(RXBN::RXB0).unwrap();
+                        defmt::info!("Received frame: {:08b}", received_frame);
+                        break;
+                    }
+                }
+                RXBN::RXB1 => {
+                    if (rx_status & (1 << 1) != 0) {
+                        defmt::info!("Frame received in RXB1");
+                        let received_frame = self.read_rx_buffer(RXBN::RXB1).unwrap();
+                        defmt::info!("Received frame: {:08b}", received_frame);
+                        break;
+                    }
+                }
+            }
+        }
+    }
+
+    /// Method for applying the CAN controller settings by getting the bitmask.
+    /// Format for setting bit: N = N | (1<<k).
+    pub fn get_canctrl_mask(&mut self, canctrl_settings: SettingsCanCtrl) -> u8 {
+        let mut canctrl_byte = 0u8; // (1): data_byte |= (reg_value << reg_pos)
+        let mode_bits: u8 = (canctrl_settings.mode as u8) << 5; //REQOP[2:0] (bits 7-5)
+        let prescale: u8 = canctrl_settings.clkpre as u8; //CLKPRE[1:0] (bits 1-0)
+        let clk_enabled = canctrl_settings.clken;
+
+        canctrl_byte |= mode_bits; //(2)
+        defmt::println!("Mode bitmask: 0b{:08b}", mode_bits);
+        defmt::println!(
+            "Mode bitmask applied to CANCTRL register: 0b{:08b}",
+            canctrl_byte
+        );
+
+        // `clippy::bool_comparision`
+        if (canctrl_settings.abat) {
+            canctrl_byte |= 1 << 4;
+        }
+
+        if (canctrl_settings.osm) {
+            canctrl_byte |= 1 << 3;
+        }
+
+        if (clk_enabled) {
+            canctrl_byte |= 1 << 2;
+        } else {
+            canctrl_byte &= !(1 << 2);
+        }
+
+        canctrl_byte |= prescale; //(3)
+        defmt::println!(
+            "3. CANCTRL bitmask after applying settings: 0b{:08b}",
+            canctrl_byte
+        );
+
+        canctrl_byte
+    }
+
+    /// Read Status Instruction.
+    ///
+    /// ## NOTE
+    ///
+    /// Will show:
+    ///
+    /// RX0IF (CANINTF[0]) - bit 0
+    /// RX1IF (CANINTF[1]) - bit 1
+    /// TXREQ (TXB0CNTRL[3]) - bit 2
+    /// TX0IF (CANINTF[2]) - bit 3
+    /// TXREQ (TXB1CNTRL[3]) - bit 4
+    /// TX1IF (CANINTF[3]) - bit 5
+    /// TXREQ (TXB2CNTRL[3]) - bit 6
+    /// TX2IF (CANINTF[4]) - bit 7
+    fn read_status(&mut self) {
+        let instruction_msg: [u8; 3] = [InstructionCommand::Status as u8, 0x00, 0x00];
+        let mut data_out: [u8; 3] = [0; 3];
+        self.cs.set_low();
+        self.spi.transfer(&mut data_out, &instruction_msg);
+        self.cs.set_high();
+        defmt::info!("Read Status: {:08b}", data_out);
```

Comment:
I think that it would be better to return a struct with the fields rather than include defmt.

This is fine for this version but for future work, i.e. when we break this out in to a distinct crate we should remove the defmt dependency

Reply:
I agree, have removed a lot of the defmt prints, but kept some for future debugging. Will remove these when we have a distinct crate in the future.